src/amo/utils/compatibility.js

```diff
@@ -311,11 +310,8 @@ export const correctedLocationForPlatform = ({
 
   if (
     // Android browser on desktop site.
-    (isFirefoxForAndroid(userAgentInfo) && clientApp === CLIENT_APP_FIREFOX) ||
-    // Android browser on page other than Home or Search.
-    (isFirefoxForAndroid(userAgentInfo) &&
-      !isHomePage &&
-      !location.pathname.includes('/search/'))
+    isFirefoxForAndroid(userAgentInfo) &&
+    clientApp === CLIENT_APP_FIREFOX
   ) {
     // Redirect to `android` home page.
     return getMobileHomepageLink(lang);
```

Comment:
Note for future code archeologists: When a Firefox Desktop browser visits android pages we build the "fixed" URL changing the application (see a few lines below), and it may be worth considering doing that in the future for Android, but it's out of scope for the moment as some pages are not supposed to be available on Android...